examples/file_body.hpp

```diff
@@ -79,7 +79,11 @@ struct file_body
         boost::tribool
         operator()(resume_context&&, error_code&, Write&& write)
         {
-            buf_len_ = std::min(size_ - offset_, sizeof(buf_));
+            if(size_ - offset_ < sizeof(buf_))
+                buf_len_ = static_cast<std::size_t>(
+                    size_ - offset_);
+            else
+                buf_len_ = sizeof(buf_);
```

Comment:
Aha. Of course, I've gotten so used to 64-bit arch, that I often forget to consider 32-bit. I need to stop doing that.